test/system_info/main.c

```diff
@@ -0,0 +1,57 @@
+#include <sys/utsname.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include "test.h"
+
+static int test_uname() {
+    struct utsname name;
+    uname(&name);
+    printf("sysname = %s\n", (const char *)&name.sysname);
+    printf("nodename = %s\n", (const char *)&name.nodename);
+    printf("release = %s\n", (const char *)&name.release);
+    printf("version = %s\n", (const char *)&name.version);
+    printf("machine = %s\n", (const char *)&name.machine);
+    printf("domainname = %s\n", (const char *)&name.__domainname);
+
+    return 0;
+}
+
+static int test_getgroups() {
+    int group_num = getgroups(0, NULL);
+    if (group_num <= 0) {
+        return -1;
```

Comment:
Please follow the established practice of using `THROW_ERROR` macro to report errors in test cases.

Reply:
Done